enterprise_access/apps/bffs/context.py

```diff
@@ -43,23 +54,203 @@ def request(self):
+        enterprise_slug_query_param = self.request.query_params.get('enterprise_customer_slug')
+
         enterprise_uuid_post_param = None
+        enterprise_slug_post_param = None
         if self.request.method == 'POST':
             enterprise_uuid_post_param = self.request.data.get('enterprise_customer_uuid')
+            enterprise_slug_post_param = self.request.data.get('enterprise_customer_slug')
 
         enterprise_customer_uuid = enterprise_uuid_query_param or enterprise_uuid_post_param
-        if enterprise_customer_uuid:
-            self.enterprise_customer_uuid = enterprise_customer_uuid
-        else:
-            raise ValueError("enterprise_customer_uuid is required for this request.")
+        self._enterprise_customer_uuid = enterprise_customer_uuid
+        enterprise_customer_slug = enterprise_slug_query_param or enterprise_slug_post_param
+        self._enterprise_customer_slug = enterprise_customer_slug
+
+        # Initialize the enterprise customer users metatata derived from the LMS
+        self._initialize_enterprise_customer_users()
+
+        if not (enterprise_customer := self.data.get('enterprise_customer')):
+            # If no enterprise customer is found, return early
+            return
+
+        # Otherwise, update the enterprise customer UUID and slug if not already set
+        if not self.enterprise_customer_slug:
+            self._enterprise_customer_slug = enterprise_customer.get('slug')
+        if not self.enterprise_customer_uuid:
+            self._enterprise_customer_uuid = enterprise_customer.get('uuid')
+
+    def _initialize_enterprise_customer_users(self):
+        """
+        Initializes the enterprise customer users for the request user.
+        """
+        try:
+            enterprise_customer_users_data = self.lms_user_api_client.get_enterprise_customers_for_user(
+                self.user.username,
+                traverse_pagination=True
+            )
+        except Exception as e:  # pylint: disable=broad-except
+            self.add_error(
+                user_message='Error retrieving linked enterprise customers',
+                developer_message=str(e)
+            )
+            return
+
+        # Set enterprise features from the response
+        self._enterprise_features = enterprise_customer_users_data.get('enterprise_features', {})
+
+        # Parse the enterprise customer user data
+        enterprise_customer_users = enterprise_customer_users_data.get('results', [])
+        active_enterprise_customer_user = next(
+            (
+                enterprise_customer_user
+                for enterprise_customer_user in enterprise_customer_users
+                if enterprise_customer_user.get('active', False)
+            ),
+            None
+        )
+        active_enterprise_customer = (
+            active_enterprise_customer_user.get('enterprise_customer')
+            if active_enterprise_customer_user else None
+        )
```

Comment:
[nit/non-blocking] This functions doing a lot. I wonder if its worth chunking up some of the logic into a separate function?
Ex. `get_enterprise_customer_users_data`, `get_active_enterprise_customer`